src/pcp/meminfo/pcp-meminfo.py

```diff
@@ -0,0 +1,195 @@
+            "mem.util.directMap2M",
+            "mem.util.directMap1G"]
+
+METRICS_DESC = ["MemTotal",
+                "MemFree",
+                "MemAvailable",
+                "Buffers",
+                "Cached",
+                "SwapCached",
+                "Active",
+                "Inactive",
+                "Active(anon)",
+                "Inactive(anon)",
+                "Active(file)",
+                "Inactive(file)",
+                "Unevictable",
+                "Mlocked",
+                "SwapTotal",
+                "SwapFree",
+                "Dirty",
+                "Writeback",
+                "AnonPages",
+                "Mapped",
+                "Shmem",
+                "Slab",
+                "SReclaimable",
+                "SUnreclaim",
+                "KernelStack",
+                "PageTables",
+                "NFS_Unstable",
+                "Bounce",
+                "WritebackTmp",
+                "CommitLimit",
+                "Committed_AS",
+                "VmallocTotal",
+                "VmallocUsed",
+                "VmallocChunk",
+                "HardwareCorrupted",
+                "AnonHugePages",
+                "ShmemHugePages",
+                "ShmemPmdMapped",
+                "CmaFree",
+                "HugePages_Total_NO_kb",
+                "HugePages_Free_NO_kb",
+                "HugePages_Rsvd_NO_kb",
+                "HugePages_Surp_NO_kb",
+                "Hugepagesize",
+                "DirectMap4k",
+                "DirectMap2M",
+                "DirectMap1G"]
+
+class MeminfoReport(pmcc.MetricGroupPrinter):
+    samples = 0
+
+    def __init__(self, samples):
+        self.samples = samples
+
+    def getMetricName(self, idx):
+        metric_name = ""
+        units = ""
+        if METRICS_DESC[idx][-6:] == "_NO_kb":
+            metric_name = METRICS_DESC[idx][:-6]
+        else:
+            metric_name = METRICS_DESC[idx]
+            units = "kB"
+        return metric_name, units
+
+    def report(self, manager):
+        group = manager["meminfo"]
+
+        opts.pmGetOptionSamples()
+
+        t_s = group.contextCache.pmLocaltime(int(group.timestamp))
+        time_string = time.strftime(MeminfoOptions.timefmt, t_s.struct_time())
+        print(time_string)
+
+        idx = 0
+        for metric in METRICS:
+            try:
+                val = group[metric].netValues[0][2]
+            except IndexError:
+                metric_name, units = self.getMetricName(idx)
+                print(F"{metric_name:17} : NA")
+
+                idx += 1
+                continue
+
+            metric_name, units = self.getMetricName(idx)
+            print(F"{metric_name:17} : {val} {units}")
+
+            idx += 1
+        print()
+
+class MeminfoOptions(pmapi.pmOptions):
+    context = None
+    timefmt = "%H:%M:%S"
+    samples = 0
+
+    def __init__(self):
+        pmapi.pmOptions.__init__(self, "a:s:S:T:z:P:A:R:t:T:x:")
```

Comment:
@mohith-kumar-thummaluru looking good!  I think some of the short options listed are unused? Definitely P:, R: and x: look like they could be dropped from the string.
Then I think we're good to merge this - thanks!

Reply:
Thanks for the comments, @natoscott. I'll filter the unused options and let you know